test/vagrant-service-manager/installer_test.rb

```diff
@@ -0,0 +1,89 @@
+require 'test_helper'
+require "vagrant/util/downloader"
+
+module VagrantPlugins
+  module ServiceManager
+    describe Installer do
+      before do
+        @type = :docker
+        @machine = fake_machine
+        @env = @machine.env
+        @options = { box_version: 'adb', '--cli-version' => '1.11.0' }
+
+        # set bin and tmp directories
+        @base_test_path = "#{@env.data_dir}/service-manager/test"
+        ServiceManager.set_bin_dir("#{@base_test_path}/bin")
+        ServiceManager.set_temp_dir("#{@base_test_path}/tmp")
+        @bin_dir = ServiceManager.bin_dir
+        @temp_dir = ServiceManager.temp_dir
+
+        # Installer instance
+        @installer = Installer.new(@type, @machine, @env, @options)
+      end
+
+      after do
+        FileUtils.rmtree(@bin_dir) if File.directory? @bin_dir
+        FileUtils.rmtree(@temp_dir) if File.directory? @temp_dir
+      end
+
+      it 'should set default values properly' do
+        @installer.instance_variable_get('@binary_exists').must_equal true
+        @installer.instance_variable_get('@url').must_equal ''
+        @installer.instance_variable_get('@archive_file_path').must_equal ''
+        @installer.instance_variable_get('@type').must_equal @type
+        @installer.instance_variable_get('@machine').must_equal @machine
+        @installer.instance_variable_get('@ui').must_equal @env.ui
+        @installer.instance_variable_get('@temp_bin_dir').must_equal "#{@temp_dir}/#{@type.to_s}"
+        @installer.instance_variable_get('@binary_name').must_equal @type.to_s
+      end
+
+      it 'should update binary path on changing' do
+        expected_path = '/usr/bin/docker'
```

Comment:
These paths will only get created on the installation(`install` method execution). Anyway for safety I will update it :+1: